app/screens/channel_info/channel_info.js

```diff
@@ -60,6 +61,7 @@ export default class ChannelInfo extends PureComponent {
         isChannelMuted: PropTypes.bool.isRequired,
         isCurrent: PropTypes.bool.isRequired,
         isFavorite: PropTypes.bool.isRequired,
+        isSystemAdmin: PropTypes.bool.isRequired,
```

Comment:
@avasconcelos114 I think you miss the fact that this action can also be performed by a team admin according to the ticket. Also you may want to validate it in the connector and pass an option like `canConvertChannel` as we do with the other options like `canManageUsers`

Reply:
That is what I initially thought as well, but it seems that both the [webapp](https://github.com/mattermost/mattermost-webapp/pull/418/files#diff-ffb3df58f36bc3727f9099bc14ff411cR737) and the [server](https://github.com/mattermost/mattermost-server/pull/8655/files#diff-e6f09296e74884d7dce741da07fc409eR157) have limited this functionality to a `MANAGE_SYSTEM` level, so if I had put that to team level in this PR the server would likely have returned a permission error.

I could definitely change this to team admin level as you pointed out, but that will require a PR to fix the server's API before this one could be merged (which would be the best option in my opinion)

Reply:
@esethna thoughts on this? the ticket mentions system/team admin, but what Andre mentions is totally true. I do agree that this should be a team/system admin feature. Maybe @wiersgallak can also give us some thought on this.

And if required we should make the changes to the server/webapp and add the version condition here.

Reply:
Ah actually, my mistake. It seems like both the webapp and the server have already been fixed to use `MANAGE_TEAM` permissions instead (the links I was following through github issues were outdated)

I'll just go ahead and replace the `isSystemAdmin` with `canConvertChannel` (based on whether the user is at least a team admin)

Thank you again for the feedback and sorry about not checking if those PRs were still valid in the current `master` branch